pkg/manifests/cache_test.go

```diff
@@ -0,0 +1,27 @@
+package manifests
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestBuildTarballURL(t *testing.T) {
+	t.Run("valid URL, no SHA", func(t *testing.T) {
+		u, err := buildTarballURL("https://example.com/foo/bar", "")
```

Comment:
modify this test case where this url already has a query param, eg `https://example.com/foo/bar?serviceId=some-id` (that's actually how this url is sent from the api)